### PR TITLE
Add repeat to spinners when holding down mouse

### PR DESCRIPTION
**Pull request by Psychpsyo**
This implements this suggestion from the discord:
image
https://discord.com/channels/811508670205788211/829768829542334544/1269740342517698690

I've implemented it in a way where the longer you hold down the mouse button, the faster it goes so that it gives fine-ish control over small adjustments while still allowing you to quickly go larger distances.

I hope just opening PRs like this is fine. (and if it is, I might do more in the future)
If not, I'm sorry.

### CODE REVIEW

**Review by tom5454 — CHANGES_REQUESTED** [review]
This won't compile on all platforms. You can't use java.util.Timer.

Store the mouse down state and update the value under the draw method.

**Comment by Psychpsyo** [comment]
> This won't compile on all platforms. You can't use java.util.Timer.
> 
> Store the mouse down state and update the value under the draw method.

Done.
I've also made it so that the repeat doesn't need to ramp up to full speed anymore.
It just kicks in at full speed after about half a second since that felt better to use.

File: CustomPlayerModels/src/shared/java/com/tom/cpl/gui/elements/Spinner.java [hunk 1]
```diff
@@ -52,31 +54,58 @@ public void draw(MouseEvent event, float partialTicks) {
 		if(txtf.isFocused() && error != null) {
 			gui.drawRectangle(bounds.x, bounds.y, bounds.w, bounds.h, 0xffff0000);
 		}
+		// mouse click repeat
+		if (currentClick != null) {
+			mouseRepeatTimer -= partialTicks;
```
Comment (tom5454) [inline CustomPlayerModels/src/shared/java/com/tom/cpl/gui/elements/Spinner.java:59]: We really shouldn't use the `partialTicks` for anything functional.
It'll break on many Minecraft versions (not even mojang knows what it should be https://bugs.mojang.com/browse/MC-273464)
On 1.21 NeoForge it's actually the partial ticks value, on Fabric and some older versions it's delta ticks from last frame,
but it changes depending on which minecraft version you are on.
And it makes the spinner buttons unusable, with high frame rates.

I'll have to go through and test every version and make sure it's the correct one (it makes the vanilla animations in the editor look jittery).

You can use:
`MinecraftClientAccess.get().getPlayerRenderManager().getAnimationEngine().getTicks()`
to get the current tick count from the game start.

**Review by tom5454 — CHANGES_REQUESTED** [review]

File: CustomPlayerModels/src/shared/java/com/tom/cpl/gui/elements/Spinner.java (hunk 1 above)
Comment (Psychpsyo) [reply CustomPlayerModels/src/shared/java/com/tom/cpl/gui/elements/Spinner.java]: Done.

File: CustomPlayerModels/src/shared/java/com/tom/cpl/gui/elements/Spinner.java [hunk 2]
```diff
@@ -52,31 +55,54 @@ public void draw(MouseEvent event, float partialTicks) {
 		if(txtf.isFocused() && error != null) {
 			gui.drawRectangle(bounds.x, bounds.y, bounds.w, bounds.h, 0xffff0000);
 		}
+		// mouse click repeat
+		if (currentClick != null && MinecraftClientAccess.get().getPlayerRenderManager().getAnimationEngine().getTicks() - currentClickStartTime > 10) {
+				arrowClicked(currentClick);
```
Comment (tom5454) [inline CustomPlayerModels/src/shared/java/com/tom/cpl/gui/elements/Spinner.java:60]: Add `currentClickStartTime++;` so it'll increment 20 times a second, not every frame.

**Review by tom5454 — APPROVED** [review]
It looks good.

File: CustomPlayerModels/src/shared/java/com/tom/cpl/gui/elements/Spinner.java (hunk 2 above)
Comment (Psychpsyo) [reply CustomPlayerModels/src/shared/java/com/tom/cpl/gui/elements/Spinner.java:61]: Done.

**Pull request merged**